src/codec.rs

```diff
@@ -0,0 +1,29 @@
+//! This module contains traits to have a unified API across codecs.
+//!
+//! There are two traits defined, [`Codec`] and [`Links`]. Those are separate traits as the `Links`
+//! trait is not generic over a certain type.
+
+use cid::Cid;
+
+/// Each IPLD codec implementation should implement this Codec trait. This way codecs can be more
+/// easily exchanged or combined.
+pub trait Codec<T> {
+    /// The multicodec code of the IPLD codec.
+    const CODE: u64;
+    /// The error that is returned if encoding or decoding fails.
+    type Error;
+
+    /// Decode a slice into the desired type.
+    fn decode(bytes: &[u8]) -> Result<T, Self::Error>;
```

Comment:
`decode_bytes`, `decode(reader)`, etc? Again with default implementations.

Well, from a nostd standpoint this is all kind of annoying. Maybe use `core2` as we do in `cid`?

Reply:
I see the `Codec` trait as a short cut for easy usage. If you want to be more advanced, use the corresponding codec directly (also the libipld-core `Codec` trait just worked on slices).

Though I'm happy to look into how hard it would be to also have it for readers.

Reply:
Personally, I see it as a way to abstract over codecs. E.g., I write some type that's generic over some codec `C` so I can let the user choose what codec they want me to use.